paraatm/simulation_method/vcas.py

```diff
@@ -0,0 +1,337 @@
+                ac.setRocd_fps(vy[ind])
+                self.simulationInterface.resume(1)
+                # check pause
+                while True:
+                    server_runtime_sim_status = self.simulationInterface.get_runtime_sim_status()
+                    # print(server_runtime_sim_status)
+                    if (server_runtime_sim_status == self.NATS_SIMULATION_STATUS_PAUSE):
+                        break
+
+                ac = self.aircraftInterface.select_aircraft(aclist[0])
+                h = ac.getAltitude_ft()
+                curr_t = self.starttime + self.simulationInterface.get_curr_sim_time()
+
+            if ind < ncmd - 2:  # the previous n-1 command
+                # print(ind)
+                if curr_t < cmd_time:
+                    # print(cmd_level)
+                    dt = cmd_time - curr_t
+                    stat = ac.getFlight_phase()
+                    ac.setCruise_alt_ft(cmd_level)
+                    ac.setFlight_phase(11)  # cruise
+                    ac.setCruise_tas_knots(tas[ind])
+
+                    self.simulationInterface.resume(dt)
+                    # check pause
+                    while True:
+                        server_runtime_sim_status = self.simulationInterface.get_runtime_sim_status()
+                        # print(server_runtime_sim_status)
+                        if (server_runtime_sim_status == self.NATS_SIMULATION_STATUS_PAUSE):
+                            break
+
+                    ac = self.aircraftInterface.select_aircraft(aclist[0])
+                    ac.setFlight_phase(stat)
+            else:  # the last command
+                stat = ac.getFlight_phase()
+                ac.setCruise_alt_ft(cmd_level)
+                ac.setFlight_phase(11)  # cruise
+                self.simulationInterface.resume(275)
+                # check pause
+                while True:
+                    server_runtime_sim_status = self.simulationInterface.get_runtime_sim_status()
+                    # print(server_runtime_sim_status)
+                    if (server_runtime_sim_status == self.NATS_SIMULATION_STATUS_PAUSE):
+                        break
+
+                ac = self.aircraftInterface.select_aircraft(aclist[0])
+                # curlon = ac.getLongitude_deg()
+                # curlat = ac.getLatitude_deg()
+                ac.setFlight_phase(stat)
+
+        self.simulationInterface.resume()
+        # check end
+        while True:
+            server_runtime_sim_status = self.simulationInterface.get_runtime_sim_status()
+            if (server_runtime_sim_status == self.NATS_SIMULATION_STATUS_ENDED):
+                break
+
+        millis = int(round(time.time() * 1000))
+        output_filename = os.path.splitext(os.path.basename(__file__))[0] + "_" + str(millis) + ".csv"
+        self.write_output(output_filename)
+        track = read_nats_output_file(output_filename)
+        self.cleanup()
+
+        return track
```

Comment:
All of this can be deleted.  With `NatsSimulationWrapper`, your `simulation` method should not create output.  That is supposed to be handled by `write_output`.  And when `NatsSimulationWrapper.__call__` calls your `simulation` method, it does not use the return value from `simulation`.  Instead, it automatically calls `write_output` and gets the return value itself.  If you remove this return value, you'll need to also make sure that you update the calls in `make_anti_model`, which are currently using it.

You should not call `cleanup` here.  It is automatically called by `__call__`.  This is why the unit test fails (cleanup is happening before it calls `write_output` to get the results).